typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -68,27 +69,30 @@ import Numeric.Natural (Natural)
 -- This runs the peer to completion (if the protocol allows for termination).
 --
 runPeer
-  :: forall ps (st :: ps) pr failure bytes m a .
-     Monad m
-  => Codec ps failure m bytes
+  :: forall ps (st :: ps) pr failure bytes m a t' .
+     MonadSTM m
+  => LogTrace t' m String
+  -> Codec ps failure m bytes
   -> Channel m bytes
   -> Peer ps pr st m a
   -> m (Either failure a)
 
-runPeer Codec{encode, decode} channel@Channel{send} = go Nothing
+runPeer tr Codec{encode, decode} channel@Channel{send} = go Nothing
   where
     go :: forall st'.
           Maybe bytes
        -> Peer ps pr st' m a
        -> m (Either failure a)
-    go trailing (Effect k) = k >>= go trailing
-    go _        (Done _ x) = return (Right x)
+    go trailing (Effect k) = traceOutput tr "Effect" >> k >>= go trailing
+    go _        (Done _ x) = traceOutput tr "Done"   >>return (Right x)
 
     go trailing (Yield stok msg k) = do
+      traceOutput tr "Yield"
       send (encode stok msg)
       go trailing k
 
     go trailing (Await stok k) = do
+      traceOutput tr "Await"
```

Comment:
You could also add modify `k` so that it logs when a message is received.

Reply:
https://github.com/input-output-hk/ouroboros-network/blob/andreas/119-logging-in-Sim/typed-protocols/src/Network/TypedProtocol/Driver.hs#L101-L103